pinot-plugins/pinot-file-system/pinot-gcs/src/main/java/org/apache/pinot/plugin/filesystem/GcsPinotFS.java

```diff
@@ -313,8 +314,12 @@ public long length(URI fileUri) throws IOException {
       }
       page.iterateAll()
           .forEach(blob -> {
-            if (!blob.getName().equals(fileUri.getPath())) {
-              builder.add(blob.getName());
+            if (!blob.getName().equals(prefix)) {
+              try {
+                builder.add(new URI(SCHEME, fileUri.getHost(), DELIMITER + blob.getName(), null).toString());
```

Comment:
Why do we need to recreate the uri?

Reply:
I think the blob is not guaranteed to has the gs URI .

Reply:
Yep, that is why, this way it's guaranteed.